spec.bs

```diff
@@ -18,3 +18,91 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="join-ad-interest-group">joinAdInterestGroup</h2>
+
+<xmp class="idl">
+[
+  SecureContext
+] partial interface Navigator {
+  Promise<undefined> joinAdInterestGroup(AuctionAdInterestGroup group, double durationSeconds);
+};
+</xmp>
+
+
+<h2 id="run-ad-auction">runAdAuction</h2>
+
+<xmp class="idl">
+[
+  SecureContext
+] partial interface Navigator {
+  Promise<USVString?> runAdAuction(AuctionAdConfig config);
+};
+</xmp>
+
+
+<xmp class="idl">
+dictionary AuctionAdConfig {
+  required USVString seller;
+  required USVString decisionLogicUrl;
+  USVString trustedScoringSignalsUrl;
+  sequence<USVString> interestGroupBuyers;
+  any auctionSignals;
+  any sellerSignals;
+  USVString directFromSellerSignals;
+  unsigned long long sellerTimeout;
+  unsigned short sellerExperimentGroupId;
+  record<USVString, any> perBuyerSignals;
+  record<USVString, unsigned long long> perBuyerTimeouts;
+  record<USVString, unsigned short> perBuyerGroupLimits;
+  record<USVString, unsigned short> perBuyerExperimentGroupIds;
+  record<USVString, record<USVString, double>> perBuyerPrioritySignals;
+  sequence<AuctionAdConfig> componentAuctions;
+  AbortSignal? signal;
+};
+</xmp>
+
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+
+<xmp class="idl">
+dictionary AuctionAd {
+  required USVString renderUrl;
+  any metadata;
+};
+</xmp>
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+
+
+<xmp class="idl">
+enum WorkletExecutionMode {
+  "compatibility",
+  "groupByOrigin",
+};
+
+dictionary AuctionAdInterestGroup {
+  required USVString owner;
+  required USVString name;
+
+  double priority;
+  boolean enableBiddingSignalsPrioritization;
+  record<USVString, double> priorityVector;
+  record<USVString, double> prioritySignalsOverrides;
+
+  WorkletExecutionMode executionMode;
+  USVString biddingLogicUrl;
+  USVString biddingWasmHelperUrl;
+  USVString dailyUpdateUrl;
+  USVString trustedBiddingSignalsUrl;
+  sequence<USVString> trustedBiddingSignalsKeys;
+  any userBiddingSignals;
+  sequence<AuctionAd> ads;
+  sequence<AuctionAd> adComponents;
+};

```

Comment:
> Put this in the same `<xmp>` block as the one that defines the function that takes it as a parameter, `joinAdInterestGroup()`. 

I thought it would be better to keep the IDL for these dictionaries separate from the functions since they can be documented separately and could then be used as parameters for more than one function.  But I'll merge them for now.